src/commands/contract/test/index.ts

```diff
@@ -0,0 +1,182 @@
+import "ts-mocha";
+import { Flags, Args } from "@oclif/core";
+import path from "node:path";
+import { globby } from "globby";
+import Mocha from "mocha";
+import { emptyDir } from "fs-extra/esm";
+import shell from "shelljs";
+import { Contract } from "../../../lib/contract.js";
+import { SwankyCommand } from "../../../lib/swankyCommand.js";
+import { ConfigError, FileError, InputError, ProcessError, TestError } from "../../../lib/errors.js";
+import { spawn } from "node:child_process";
+import { Spinner } from "../../../lib/index.js";
+
+declare global {
+  var contractTypesPath: string; // eslint-disable-line no-var
+}
+
+export class TestContract extends SwankyCommand<typeof TestContract> {
+  static description = "Run tests for a given contact";
+
+  static flags = {
+    all: Flags.boolean({
+      default: false,
+      char: "a",
+      description: "Set all to true to compile all contracts",
+    }),
+    mocha: Flags.boolean({
```

Comment:
Not sure about this flag, this initial idea was to define it during init:

> I suggest in the init phase, if ink! is selected, only bootstrap ink! e2e, and mocha for ask!

Reply:
@ipapandinas  Am I understand right?  We want to pick what type of tests we will use when we add a contract or init project.

Reply:
We can preserve the `--mocha` (check the global comment for UXcontext). However perform `generateTypes` in that case.